src/main/java/com/pscheduler/server/model/CourseGPA.java

```diff
@@ -0,0 +1,176 @@
+package com.pscheduler.server.model;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Version;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+@Entity
+public class CourseGPA {
+
+    @Id
+    @Min(value = 10000)
+    @Max(value = 99999)
+    private int crn;
+
+    @NotNull
+    @Size(min = 2, max = 6)
+    private String subject;
+
+    @NotNull
+    @Size(min = 4, max = 5)
+    private String courseNumber;
+
+    @NotNull
+    @Size(min = 1, max = 100)
+    private String name;
+
+    @NotNull
+    @Size(min = 1, max = 50)
+    private String instructor;
+
+    @Min(value = 0)
+    @Max(value = 50)
+    private int credits;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 4)
+    private double gpa;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 1000)
+    private int students;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 100)
+    private double A;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 100)
+    private double B;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 100)
+    private double C;
+
+    @Min(value = 0)
+    @Max(value = 100)
+    private double D;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 100)
+    private double F;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 1000)
+    private int withdraws;
+
+    @Version
+    private long version;
+
+    public CourseGPA() {
+    }
+
+    public CourseGPA(
+        String subject,
+        String courseNumber,
+        String name,
+        String instructor,
+        int crn,
+        int credits,
+        double gpa,
+        int students,
+        double A,
+        double B,
+        double C,
+        double D,
+        double F,
+        int withdraws
+    ) {
+        this.subject = subject;
+        this.courseNumber = courseNumber;
+        this.name = name;
+        this.instructor = instructor;
+        this.crn = crn;
+        this.credits = credits;
+        this.gpa = gpa;
+        this.students = students;
+        this.A = A;
+        this.B = B;
+        this.C = C;
+        this.D = D;
+        this.F = F;
+        this.withdraws = withdraws;
+    }
+
+    public String getSubject() {
+        return this.subject;
+    }
+
+    public void setSubject(String subject) {
+        this.subject = subject;
+    }
+
+    public String getCourseNumber() { return this.courseNumber; }
```

Comment:
also lets put the variable assignment / returns in a newline

Reply:
Fixed